### PR TITLE
cleanup and better integration with other mods

### PR DESCRIPTION
**Pull request by fluxionary**
the motivation for this PR was to fix a bug, whereby a player attached to an object can sit in a chair/bench, and then somehow gain the ability to "fly" horizontally. you can see this w/ the minecart from minetest_game: get in the cart, sit in the chair (you'll remain in the cart), pick up the cart. at this point, the effect should be active. you can also trigger this glitch if you sit on a chair, then use e.g. the `/sit` command from the [emote mod](https://github.com/minetest-mods/emote). 

* removed deprecated metadata files
* added luacheck workflow
* remote hard dependency on "default" (minetest_game)
* prevent sitting in a chair if a player is attached to something
* use player monoids to manage physics and attachment state if available

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -7,9 +7,13 @@ ts_furniture.kneeling_bench = minetest.settings:get_bool("ts_furniture.kneeling_
 
 -- Used for localization
 local S = minetest.get_translator("ts_furniture")
+local has_more_player_monoids = minetest.get_modpath("more_player_monoids")
+local has_player_monoids = minetest.get_modpath("player_monoids")
+
+local is_sitting_by_player_name = {}
```
Comment (SmallJoker) [inline init.lua:13]: Needs removal upon player leave

**Comment by Niklp09** [comment]
@fluxionary do you plan to continue work on this, or can someone (me) adopt it?

**Comment by fluxionary** [comment]
> @fluxionary do you plan to continue work on this, or can someone (me) adopt it?

i lost track of this and the replies. AliasAlreadyTaken from your-land poked me about it just now. putting it on my todo list for tomorrow.

File: init.lua (hunk 1 above)
Comment (fluxionary) [reply init.lua:13]: i missed this last year, and someone just poked me to fix it now. anyone got a good way to create recurring reminders for such things?

Comment (SmallJoker) [reply init.lua:13]: Perhaps check the status of your PRs at https://github.com/pulls every once in a while?

**Comment by Niklp09** [comment]
Closing since #22 has been merged. As always, reopen if needed.

**Pull request closed without merging**